Add examples for ORD() (#16956)
pingcap/docs: functions-and-operators/string-functions.md

The `CHARACTER_LENGTH()` function is the same as the `CHAR_LENGTH()` function. Both functions can be used synonymously because they generate the same output.

> ### [`CONCAT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_concat)

Author: @Oreoxmt @qiancai this is a somewhat unrelated bugfix.

### [`OCTET_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_octet-length)

Synonym for [`LENGTH()`](#length).

### [`ORD()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ord)

Return the character code for the leftmost character of the given argument.

This function is similar to [`CHAR()`](#char) but works the other way around.

Examples:

Taking `a` and `A` as an example, `ORD()` returns `97` for `a` and `65` for `A`.

```sql
SELECT ORD('a'), ORD('A');
```

```
+----------+----------+
| ORD('a') | ORD('A') |
+----------+----------+
|       97 |       65 |
+----------+----------+
1 row in set (0.00 sec)
```

If you take the character code obtained from `ORD()` as input, you can get the original characters back using the `CHAR()` function. Note that the output format might vary depending on whether the `binary-as-hex` option is enabled in your MySQL client.

```sql
SELECT CHAR(97), CHAR(65);
```

```
+----------+----------+
| CHAR(97) | CHAR(65) |
+----------+----------+
| a        | A        |
+----------+----------+
1 row in set (0.01 sec)
```

> The following example shows how `ORD()` handles multibyte characters. Here, both `101` and `0x65` have the UTF-8 encoded value for the `e` character, but in different formats. And both `50091` and `0xC3AB` have the same value, but for the `ë` character. 

Reviewer: ```suggestion
The following example shows how `ORD()` handles multibyte characters. Here, both `101` and `0x65` have the same UTF-8 encoded value for the `e` character, but in different formats. And both `50091` and `0xC3AB` have the same value, but for the `ë` character. 
```

### [`OCTET_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_octet-length)

Synonym for [`LENGTH()`](#length).

> ### [`ORD()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_ord)

Reviewer 2: Side note, should we still point to the MySQL documentation, when we have our own description?

Author: As I said in https://github.com/pingcap/docs/issues/15338#issuecomment-2039767839 I think that should be done at a later stage after our own docs have been tested a bit. Then eventually we can move the links to the MySQL Reference Manual to the "See also" section.

> Return character code for leftmost character of the argument.
> 
> This function is similar to [`CHAR()`](#char), but works the other way around.

Reviewer: ```suggestion
Return the character code for the leftmost character of the given argument.

This function is similar to [`CHAR()`](#char) but works the other way around.
```

Examples:

> Here we take "a" and "A" and use `ORD()` to get "97" and "65".

Reviewer: ```suggestion
Taking `a` and `A` as an example, `ORD()` returns `97` for `a` and `65` for `A`.
```

```sql
SELECT ORD('a'), ORD('A');
```

```
+----------+----------+
| ORD('a') | ORD('A') |
+----------+----------+
|       97 |       65 |
+----------+----------+
1 row in set (0.00 sec)
```

> If we take the octal numbers we got from `ORD()` then we can get back to the orignal characters by using the `CHAR()` function. Note that the display of the output might be different if your MySQL client has the `binary-as-hex` option enabled.

Reviewer: ```suggestion
If you take the character code obtained from `ORD()` as input, you can get the original characters back using the `CHAR()` function. Note that the output format might vary depending on whether the `binary-as-hex` option is enabled in your MySQL client.
```

```sql
SELECT CHAR(97), CHAR(65);
```

```
+----------+----------+
| CHAR(97) | CHAR(65) |
+----------+----------+
| a        | A        |
+----------+----------+
1 row in set (0.01 sec)
```

> In the example below you can see how `ORD()` handles multibyte characters. Here 101 and 0x65 are both the value for the "e" character, but in different formats. And 50091 and 0xC3AB are the same, but for the "ë" character. 

Reviewer 2: Should we be more clear with the examples of `ë` is the hex representation of the UTF-8 bytes? (not the unicode code point)

> In the example below you can see how `ORD()` handles multibyte characters. Here 101 and 0x65 are both the UTF-8 encoded value for the "e" character, but in different formats. And 50091 and 0xC3AB are also the same value, but for the "ë" character. 

Reviewer: ```suggestion
The following example shows how `ORD()` handles multibyte characters. Here, both `101` and `0x65` have the UTF-8 encoded value for the `e` character, but in different formats. And both `50091` and `0xC3AB` have the same value, but for the `ë` character. 
```
